exercises/binary_converter.cpp

```diff
@@ -8,19 +8,23 @@
 #include <iostream>
 #include <string>
 
-int main() {
-    std::string num_tmp;
-    int n, lenght;
-    std::cin >> n;
 
+std::string binaryConv( int n){
+    std::string n_tmp;
     while(n != 0){
-        if(n%2 == 0){
-            num_tmp += '0';
-        }else {
-            num_tmp += '1';
-        }
+        n_tmp += n % 2 ? "1" : "0";
         n = n/2;
```

Comment:
`? "1" : "0"` è un po' come `? true : false`, con qualche trick forse potresti abbreviarlo

Reply:
Abbreviare tutta l'istruzione o solo  `n_tmp += n % 2 ? "1" : "0";`

Reply:
`n_tmp += to_string(n % 2);`

perchè n%2 restituisce o true o false, quindi o 1 o 0, se converti quell'1 o 0 in "stringa" eviti di fare `? "1" : "0"`

non so se al livello di efficienza è meglio la condizione ternaria rispetto al casting to_string(), però è un'alternativa 👀

Reply:
Non sapevo che il modulo dava true e false pensavo che restituiva solo il resto della divisione, da quello che ho provato sembra che ci mettano circa lo stesso tempo a compilare e ad eseguire pero sono sempre considerazioni molto superficiali

Reply:
ups, scusa l'ho scritto di fretta, non restituisce o true o false, ma nel caso di "2" può restituire o 1 o 0 (paragonabili a true o false) che puoi convertire in stringa

n.b.: true o false si possono spesso vedere come 1 o 0.